pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -0,0 +1,191 @@
+import fs from 'fs';
+import os from 'os';
+import { join } from 'path';
+
+import * as nativeReg from 'native-reg';
+import plist from 'plist';
+
+import * as settings from '@pkg/config/settings';
+import Logging from '@pkg/utils/logging';
+import paths from '@pkg/utils/paths';
+
+const console = Logging.settings;
+
+const REGISTRY_PATH_PROFILE = ['SOFTWARE', 'Rancher Desktop', 'Profile'];
+
+/**
+ * Lockable default settings used for validating deployment profiles.
+ * Data values are ignored, but types are used for validation.
+ */
+const lockableDefaultSettings = {
+  containerEngine: {
+    imageAllowList: {
+      enabled:  true,
+      patterns: [] as Array<string>,
+    },
+  },
+};
+
+/**
+ * Read and validate deployment profiles, giving system level profiles
+ * priority over user level profiles.  If the system directory contains a
+ * defaults or locked profile, the user directory will not be read.
+ * @returns type validated defaults and locked deployment profiles, and throws
+ *          an error if there is an error parsing the locked profile.
+ * NOTE: The renderer process can not access the 'native-reg' library, so the
+ *       win32 portions of the deployment profile reader functions must be
+ *       located in the main process.
```

Comment:
Honestly, the renderer shouldn't know anything about how we load profiles anyway (the fact that the registry / `/etc/` / etc. was involved is an implementation detail).